File: ocaml/difference-of-squares/difference_of_squares.ml
let square_of_sum n = let sum = n * (n + 1) / 2 in sum * sum 
let sum_of_squares n = n * (n + 1) * (2 * n +1) / 6 (* https://en.wikipedia.org/wiki/Square_pyramidal_number *)
let difference_of_squares n = square_of_sum n - sum_of_squares n

File: julia/difference-of-squares/difference-of-squares.jl
square_of_sum(n::Int) = div(n * (n + 1), 2)^2
sum_of_squares(n::Int) = div(n * (n + 1) * (2 * n + 1), 6)
difference(n::Int) = square_of_sum(n) - sum_of_squares(n)

